File: ios/Runner/Subdivision.swift
let locationPrecision: Float = 2

class Subdivision {
  var option: Int
  var volume: Float
  
  init(_ subdivisionAsJson: [String: Any]) {
    self.option = subdivisionAsJson["option"] as! Int
    self.volume = Float(subdivisionAsJson["volume"] as! Double)
  }
  
  func getLocations() -> [Float] {
    var startFrames: [Float] = Array(repeating: 1.0, count: self.option - 1)
      
    for (index, startFrame) in startFrames.enumerated() {
      let fullLocation: Float = startFrame / Float(self.option) * Float(index + 1)
      startFrames[index] = (fullLocation * pow(10, locationPrecision)).rounded() / pow(10, locationPrecision)
    }
      
    return startFrames
  }
}

File: android/app/src/main/kotlin/com/lvnlx/metronomic/Subdivision.kt
package com.lvnlx.metronomic

import kotlin.math.pow
import kotlin.math.roundToInt

class Subdivision(var option: Int, var volume: Float) {
    val locationPrecision: Double = 2.0

    override fun toString(): String {
        return "Subdivision(option: ${option}, volume ${volume})"
    }

    fun getLocations(): Array<Float> {
        var startFrames: Array<Float> = Array(option - 1) { 1f }

        for ((index, startFrame) in startFrames.withIndex()) {
            val fullLocation: Float = startFrame / option * (index + 1)
            startFrames[index] = (fullLocation * 10.0.pow(locationPrecision)).roundToInt() / 10.0.pow(locationPrecision).toFloat()
        }

        return startFrames
    }
}